libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -808,7 +824,7 @@ void PbftManager::identifyBlock_() {
   auto [round, period] = getPbftRoundAndPeriod();
   LOG(log_dg_) << "PBFT filtering state in round: " << round << ", period: " << period;
 
-  if (round == 1 || next_votes_manager_->haveEnoughVotesForNullBlockHash()) {
+  if (round == 1 || previous_round_next_voted_null_block_hash_) {
```

Comment:
Same thing here. Shouldn't other voted value(not NULL_HASH) be prioritized here?

Reply:
You mean in terms of code execution, not in terms of algorithm logic?

Reply:
this condition does exactly as in paper: 
```
If p = 1 OR (p ≥ 2) AND (i has received 2t + 1 next-votes for ⊥ for period p − 1) ,
then i identifies his leader ` i,p for period p and soft-votes the value v proposed by ` i,p ;
```

previous_round_next_voted_null_block_hash_ is set to true if we saw 2t+1 next votes for NULL_BLOCK_HASH in previous round